### PR TITLE
Clearer separation of tests & application logic

### LINKED ISSUE #121: Clearer separation of tests & application logic
Followup from #112 

See comments from @robert-zaremba in his review [here](https://github.com/regen-network/regen-ledger/pull/112#discussion_r515403959)

### PR DESCRIPTION
**Pull request by aleem1413**
Fixes #121 

Feel free to submit a Draft Pull Request as soon as you start working on something. Before you mark your PR as Ready For Review, make sure you've checked off the following boxes or indicated N/A (not applicable):
- [ ] linked to a relevant issue or have written a clear description of the issue in the PR
- [ ] wrote or updated tests
- [ ] wrote or updated documentation
- [ ] added an appropriate entry in `CHANGELOG.md` following https://keepachangelog.com

Thanks!

### CODE REVIEW

**Review by anilCSE — APPROVED** [review]
Lgtm

File: app/app.go [hunk 1]
```diff
@@ -81,8 +81,6 @@ import (
 	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
```
Comment (amaurymartiny) [inline app/app.go:81]: In this file, we can also remove `testdata.RegisterQueryServer(app.GRPCQueryRouter(), testdata.QueryImpl{})`

**Review by amaurymartiny — COMMENTED** [review]
Looks good until now.

I have further suggestions:
- `app/regen/cmd/genaccounts.go`: Can we remove this file and use simapp's one?
- `app/regen/cmd/testnet.go`: I wasn't aware of this command. Do we really need it? I would propose to remove it, and use docker-compose if needed.
- `app/encoding.go`: We use `params.MakeTestEncodingConfig()`. The idea of renaming to Test was to tell users not to use this one. Maybe we should not call this function, and just copy/paste its content explicitly? Alternatively, in cosmos-sdk, export a `MakeProtoEncodingConfig()` maybe inside `std/`?

**Comment by aleem1413** [comment]
> * `app/regen/cmd/genaccounts.go`: Can we remove this file and use simapp's one?
Yes, we can reuse it from the simapp.

**Comment by anilCSE** [comment]
> Looks good until now.
> 
> I have further suggestions:
> 
> * `app/regen/cmd/genaccounts.go`: Can we remove this file and use simapp's one?

Using simapp's genaccounts might not be a good choice. I would suggest to keep it here, thought it looks like a copy-paste. 

> * `app/regen/cmd/testnet.go`: I wasn't aware of this command. Do we really need it? I would propose to remove it, and use docker-compose if needed.

Yes, we can remove this

> * `app/encoding.go`: We use `params.MakeTestEncodingConfig()`. The idea of renaming to Test was to tell users not to use this one. Maybe we should not call this function, and just copy/paste its content explicitly? Alternatively, in cosmos-sdk, export a `MakeProtoEncodingConfig()` maybe inside `std/`?

++

**Comment by aleem1413** [comment]
> > Looks good until now.
> > I have further suggestions:
> > 
> > * `app/regen/cmd/genaccounts.go`: Can we remove this file and use simapp's one?
> 
> Using simapp's genaccounts might not be a good choice. I would suggest to keep it here, though it looks like a copy-paste.
> 
> > * `app/regen/cmd/testnet.go`: I wasn't aware of this command. Do we really need it? I would propose to remove it, and use docker-compose if needed.
> 
> Yes, we can remove this
> 
> > * `app/encoding.go`: We use `params.MakeTestEncodingConfig()`. The idea of renaming to Test was to tell users not to use this one. Maybe we should not call this function, and just copy/paste its content explicitly? Alternatively, in cosmos-sdk, export a `MakeProtoEncodingConfig()` maybe inside `std/`?
> 
> ++
> 
> Edit:
> I feel like we should use simapp only for testing and avoid it in the app. I would suggest to have our own `app/params` package with following files:
> 
> * `protocodec.go`
> * `encoding.go`
> 
> `encoding.go` will have our own `EncodingConfig`, something like below
> 
> ```go
> // EncodingConfig specifies the concrete encoding types to used for regen app.
> type EncodingConfig struct {
> 	InterfaceRegistry types.InterfaceRegistry
> 	Marshaler         codec.Marshaler
> 	TxConfig          client.TxConfig
> 	Amino             *codec.LegacyAmino // or even nil may be?
> }
> ```
> 
> `protocodec.go`
> 
> ```go
> // MakeEncodingConfig creates an EncodingConfig
> func MakeEncodingConfig() EncodingConfig {
> 	amino := codec.NewLegacyAmino()
> 	interfaceRegistry := types.NewInterfaceRegistry()
> 	marshaler := codec.NewProtoCodec(interfaceRegistry)
> 	txCfg := tx.NewTxConfig(marshaler, tx.DefaultSignModes)
> 
> 	return EncodingConfig{
> 		InterfaceRegistry: interfaceRegistry,
> 		Marshaler:         marshaler,
> 		TxConfig:          txCfg,
> 		Amino:             amino, //may be nil?
> 	}
> }
> ```
> 
> @amaurymartiny wdyt?

This looks better.

File: app/params/encoding.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package params
+
+import (
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+)
+
+// EncodingConfig specifies the concrete encoding types to used for regen app.
+type EncodingConfig struct {
```
Comment (amaurymartiny) [inline app/params/encoding.go:10]: while removing the `MakeTestEncodingConfig` is desired, I believe copy/pasting this struct is not needed: we can use simapp's one directly.

**Review by amaurymartiny — APPROVED** [review]
last small comment

**Review by robert-zaremba — APPROVED** [review]
Good cleaning.

**Comment by robert-zaremba** [comment]
BTW, I really like the [band chain](https://github.com/bandprotocol/bandchain/tree/master/chain) code organization for app and cmd. It is same as a standard go packages:
* app is lean
* main packages are directly in `/cmd`

**Pull request merged**